src/hu/elte/txtuml/stdlib/math/BigIntegerImplementation.java

```diff
@@ -0,0 +1,398 @@
+	 *         tested for primality.
+	 * @throws ArithmeticException {@code bitLength < 2} or {@code bitLength} is too large.
+	 * @see    #bitLength()
+	 */
+	public BigIntegerImplementation(int bitLength, int certainty, Random rnd) {
+		this.num = new java.math.BigInteger(bitLength, certainty, rnd);
+	}
+
+	/**
+	 * Constructs a randomly generated BigInteger, uniformly distributed over
+	 * the range 0 to (2<sup>{@code numBits}</sup> - 1), inclusive.
+	 * The uniformity of the distribution assumes that a fair source of random
+	 * bits is provided in {@code rnd}.  Note that this constructor always
+	 * constructs a non-negative BigInteger.
+	 *
+	 * @param  numBits maximum bitLength of the new BigInteger.
+	 * @param  rnd source of randomness to be used in computing the new
+	 *         BigInteger.
+	 * @throws IllegalArgumentException {@code numBits} is negative.
+	 * @see #bitLength()
+	 */
+	public BigIntegerImplementation(int numBits, Random rnd) {
+		this.num = new java.math.BigInteger(numBits, rnd);
+	}
+
+	/**
+	 * Translates the decimal String representation of a BigInteger into a
+	 * BigInteger.  The String representation consists of an optional minus
+	 * sign followed by a sequence of one or more decimal digits.  The
+	 * character-to-digit mapping is provided by {@code Character.digit}.
+	 * The String may not contain any extraneous characters (whitespace, for
+	 * example).
+	 *
+	 * @param val decimal String representation of BigInteger.
+	 * @throws NumberFormatException {@code val} is not a valid representation
+	 *         of a BigInteger.
+	 * @see    Character#digit
+	 */
+	public BigIntegerImplementation(String val) {
+		this.num = new java.math.BigInteger(val);
+	}
+
+	/**
+	 * Translates the String representation of a BigInteger in the
+	 * specified radix into a BigInteger.  The String representation
+	 * consists of an optional minus or plus sign followed by a
+	 * sequence of one or more digits in the specified radix.  The
+	 * character-to-digit mapping is provided by {@code
+	 * Character.digit}.  The String may not contain any extraneous
+	 * characters (whitespace, for example).
+	 *
+	 * @param val String representation of BigInteger.
+	 * @param radix radix to be used in interpreting {@code val}.
+	 * @throws NumberFormatException {@code val} is not a valid representation
+	 *         of a BigInteger in the specified radix, or {@code radix} is
+	 *         outside the range from {@link Character#MIN_RADIX} to
+	 *         {@link Character#MAX_RADIX}, inclusive.
+	 * @see    Character#digit
+	 */
+	public BigIntegerImplementation(String val, int radix) {
+		this.num = new java.math.BigInteger(val, radix);
+	}
+
+	private java.math.BigInteger num;
+
+	@Override
+	public BigInteger abs() {
+		return new BigIntegerImplementation(num.abs());
+	}
+
+	@Override
+	public BigInteger add(BigInteger val) {
+		return new BigIntegerImplementation(num.add(getOrigBigInteger(val)));
```

Comment:
I see. You can either change the signature or cast `val` to `BigIntegerImplementation` in the parameter. It would be nice if we could get rid of the `getOrigBigInteger` function. Maybe @devaigergely81 can suggest other options.